crates/types/src/operator_metadata.rs

```diff
@@ -0,0 +1,344 @@
+use alloy::transports::http::reqwest::{self, Url};
+use mime_sniffer::MimeTypeSniffer;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use thiserror::Error;
+
+/// OperatorMetadata is the metadata operator uploads while registering
+/// itself to EigenLayer.
+#[derive(Deserialize, Serialize)]
+pub struct OperatorMetadata {
+    /// Name of the operator
+    name: String,
+
+    /// Description of the operator. There is a 200-character limit
+    description: String,
+
+    /// Logo of the operator. This should be a link to a image file
+    /// which is publicly accessible
+    logo: String,
+
+    /// Website of the operator
+    website: Option<String>,
+
+    /// Twitter handle of the operator (optional)
+    twitter: Option<String>,
+}
+
+#[derive(Debug, Error, PartialEq, Eq, Clone, Copy)]
+pub enum OperatorMetadataError {
+    #[error("Name cannot be empty")]
+    NameEmpty,
+    #[error("Name has to be less than 500 characters long")]
+    NameTooLong,
+    #[error("Name contains invalid characters")]
+    NameInvalid,
+
+    #[error("Description cannot be empty")]
+    DescriptionEmpty,
+    #[error("Description has to be less than 500 characters long")]
+    DescriptionTooLong,
+    #[error("Description contains invalid characters")]
+    DescriptionInvalid,
+
+    #[error("Logo cannot be empty")]
+    LogoUrlEmpty,
+    #[error("Logo url is invalid")]
+    LogoUrlInvalid,
+    #[error("Logo url has an invalid image extension")]
+    LogoUrlInvalidImageExtension,
+    #[error("Logo has an unsupported mime type")]
+    LogoUrlInvalidMimeType,
+
+    #[error("Website url is invalid")]
+    WebsiteUrlInvalid,
+    #[error("Website url is longer than 1024 characters")]
+    WebsiteUrlTooLong,
+    #[error("Website url points to local server")]
+    WebsiteUrlPointsToLocalServer,
+
+    #[error("Twitter url is invalid. It must be of the format https://twitter.com/<username> or https://x.com/<username>")]
+    TwitterUrlInvalid,
+    #[error("Twitter url is longer than 1024 characters")]
+    TwitterUrlTooLong,
+    #[error("Twitter url points to local server")]
+    TwitterUrlPointsToLocalServer,
+}
+
+impl OperatorMetadata {
+    pub async fn validate(&self) -> Result<(), OperatorMetadataError> {
+        // Alias the error types for brevity
+        use OperatorMetadataError::*;
+
+        let text_regex = regex::Regex::new(r#"^[a-zA-Z0-9 +.,;:?!'’"“”\-_/()\[\]~&#$—%]+$"#)
+            .expect("regex is valid");
```

Comment:
is  expect expected ?

Reply:
Yes. I can add a test to check this is so.

Reply:
`Regex::new` fails only if the regex pattern is invalid: https://docs.rs/regex/latest/regex/struct.Regex.html#method.new